### PR TITLE
add VCloudDirectorComputeServiceLiveTest

### PR DESCRIPTION
**Pull request by andreaturli**

### CODE REVIEW

**Comment by andreaturli** [comment]
@nakomis this test should track VCOMPOSE-909 (delete node does not deleting vAPP). it is always green for me.

File: src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceLiveTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      ListeningExecutorService userExecutor = context.utils().injector()
+              .getInstance(Key.get(ListeningExecutorService.class, Names.named(PROPERTY_USER_THREADS)));
+
+      try {
+         template = buildTemplate(client.templateBuilder().imageNameMatches("cloudsoft-template"));
+         template.getOptions().inboundPorts(22);
+         nodes.addAll(client.createNodesInGroup(group, 2, template));
+      } finally {
+         for (final NodeMetadata node : nodes) {
+            ListenableFuture<Boolean> future = userExecutor.submit(new Callable<Void>() {
```
Comment (aledsage) [inline src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceLiveTest.java]: @andreaturli travis-ci is right: this should be `ListenableFuture<Void>` (or could use `<?>`). And change the declaration of `futures`.

File: src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceLiveTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+               public Void call() throws Exception {
+                  getAnonymousLogger().info("Destroying node " + node.getId());
+                  client.destroyNode(node.getId());
+               }
+            });
+            futures.add(future);
+         }
+         Futures.allAsList(futures).get();
+         assertTrue(client.listNodesDetailsMatching(new Predicate<ComputeMetadata>() {
```
Comment (aledsage) [inline src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceLiveTest.java:79]: Will this assert that the VMs have been deleted, or also that the vApps were deleted? I believe the report from the customer said that empty vApps (i.e. vApps containing no VMs) were left behind.

Comment (andreaturli) [inline src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceLiveTest.java:79]: A jclouds node is mapped on a vCD VM, listNodes (and listNodeDetailsMatching) will return the list of VMs. An empty vApp could not be destroyed as it was probably created by other processes/manually. wdyt?

Comment (andreaturli) [inline src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceLiveTest.java:79]: Real problem is that listNodes was not yet implemented. Adding it now

File: src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java [hunk 3]
```diff
@@ -330,7 +333,17 @@ public VAppTemplate getImage(final String imageId) {
 
    @Override
    public Iterable<Vm> listNodes() {
-      return Sets.newHashSet();
+      Set<Vm> vms = Sets.newHashSet();
```
Comment (aledsage) [inline src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java]: Personal preference for `newLinkedHashSet` as gives more determinism - same order as they are returned to us by the vcloud api.

File: src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java [hunk 4]
```diff
@@ -330,7 +333,17 @@ public VAppTemplate getImage(final String imageId) {
...
+      QueryResultRecords result = api.getQueryApi().vmsQueryAll();
+      for (QueryResultRecordType record : result.getRecords()) {
+         if (record instanceof QueryResultVMRecord) {
+            QueryResultVMRecord queryResultVMRecord = (QueryResultVMRecord) record;
+            if (!queryResultVMRecord.isVAppTemplate()) { // && queryResultVMRecord.getStatus().equals("POWERED_ON")) {
```
Comment (aledsage) [inline src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java]: Why the commented out code? I presume we want all nodes, not just the powered on ones so delete that?

File: src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceLiveTest.java (hunk 2 above)
Comment (aledsage) [inline src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceLiveTest.java:79]: Not sure I follow your comment: "an empty vApp could not be destroyed as it was probably created by other processes/manually". 

The customer hasn't been able to reproduce the problem, so we don't know what caused the empty vApp to be left behind. But the vApp name suggested it was created by Brooklyn. It's possible that someone had deleted the VM manually, though that seems less likely.

What I was thinking of specifically for the test is an assertion that the vApp has been deleted (as well as asserting that the VM is no longer there).

File: src/test/java/org/jclouds/vcloud/director/v1_5/domain/dmtf/EnvelopeTest.java [hunk 5]
```diff
@@ -39,7 +39,7 @@ protected void setup() throws JAXBException {
    public void testUnmarshallEnvelope() throws JAXBException {
       Unmarshaller unmarshaller = jc.createUnmarshaller();
       unmarshaller.setEventHandler(new javax.xml.bind.helpers.DefaultValidationEventHandler());
-      File xml = new File("vcloud-director/src/test/resources/dmtf/envelope.xml");
+      File xml = new File("src/test/resources/dmtf/envelope.xml");
```
Comment (aledsage) [inline src/test/java/org/jclouds/vcloud/director/v1_5/domain/dmtf/EnvelopeTest.java]: Is this the normal way jclouds gets the xml from a resource in `src/test/resources`?

If so, then fine. But my guess was that the norm would be to load the resource `dmtf/envelope.xml`, and then to get its contents.

**Comment by andreaturli** [comment]
closing it as it was against a deprecated branch (1.8.x)

**Pull request closed without merging**